File: backend/app/schemas/analytics.py
"""Pydantic schemas for the analytics router."""

import uuid
from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel


class SnapshotResponse(BaseModel):
    id: uuid.UUID | None = None
    snapshot_date: date
    total_customers: int
    active_customers: int
    churned_this_period: int
    churn_rate: Decimal | None = None
    mrr: Decimal | None = None
    mrr_at_risk: Decimal | None = None
    avg_churn_probability: Decimal | None = None
    critical_risk_count: int
    high_risk_count: int
    medium_risk_count: int
    low_risk_count: int
    created_at: datetime | None = None

    model_config = {"from_attributes": True}


class SnapshotListResponse(BaseModel):
    snapshots: list[SnapshotResponse]
    total: int

File: frontend/src/api/analytics.ts
import api from './client'

export interface AnalyticsSummary {
  id: string | null
  snapshot_date: string
  total_customers: number
  active_customers: number
  churned_this_period: number
  churn_rate: string | null
  mrr: string | null
  mrr_at_risk: string | null
  avg_churn_probability: string | null
  critical_risk_count: number
  high_risk_count: number
  medium_risk_count: number
  low_risk_count: number
  created_at: string | null
}

export const analyticsApi = {
  summary: () =>
    api.get<AnalyticsSummary>('/api/v1/analytics/summary'),

  createSnapshot: () =>
    api.post<AnalyticsSummary>('/api/v1/analytics/snapshot'),
}
